src/avram/upsert.cr

```diff
@@ -90,6 +90,29 @@ module Avram::Upsert
     end
   end
 
+  macro upsert_unique_on(*attribute_names)
+    def self.upsert(upserts : Array(X)) forall X
+      \{%
+        if X > NamedTuple
+          raise("All array elements for #{@type}.upsert must be NamedTuples. Given: #{X}")
```

Comment:
```suggestion
          raise("All array elements for #{@type}.upsert must be NamedTuples. You provided: #{X}")
```

I'm not sure what comparative exceptions around Avram might be. "You provided:" isn't my favorite but "Given:" feels like it doesn't lead me in the right direction. Similarly below.

Reply:
Makes sense, I didn’t love Given either but was aiming for terseness - “you provided” def reads better